docker/pravega/scripts/init_segmentstore.sh

```diff
@@ -0,0 +1,20 @@
+#!/bin/sh
+#
+# Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+
+init_segmentstore() {
+    [ ! -z "$PUBLISHED_ADDRESS" ] && add_system_property "pravegaservice.publishedIPAddress" "${PUBLISHED_ADDRESS}"
+    [ ! -z "$PUBLISHED_ADDRESS" ] && add_system_property "pravegaservice.listeningIPAddress" "0.0.0.0"
```

Comment:
@adrianmo you argued previously that this change is necessary to work in a k8s env, but this change is being applied generally, independent of k8s. Is it a bug that we need to fix? I'm actually thinking that we might want to have multiple endpoints for internal and external connectivity, but that would be a change to the segment store.

Reply:
Actually this is a new feature that was not available before, i.e., the ability to update the published IP and port for the Segment Store. It is used in K8 w/ external connectivity, but it can also be leveraged in other environments like Docker Swarm, DCOS, or any other platform where Pravega is running behind a firewall/load balancer and is not directly accessible by clients.